services/controllerhost/consumer.go

```diff
@@ -511,7 +510,31 @@ func selectNextExtentsToConsume(
 		}
 	}
 
-	return result, nAvailable, nil
+	if len(cgExtents.openHealthy) == 0 && len(result) == 0 {
+
+		nBacklog := nAvailable + len(cgExtents.open)
+		maxExtentsToConsume := maxExtentsToConsumeForDst(context, dstDesc.GetPath(), cgDesc.GetConsumerGroupName(), getDstType(dstDesc), dstDesc.GetZoneConfigs())
+
+		if nBacklog < maxExtentsToConsume {
+			// No consumable extents for this destination, create one
+			extentID, _, storehosts, e := createExtent(context, dstID, dstDesc.GetIsMultiZone(), m3Scope)
```

Comment:
can you move this logic into a helper ? controller tries to keep method length <= 100 at max.